test/unit_tests/cellular/cellular.cpp

```diff
@@ -53,33 +53,45 @@ class FakeStream: public Print {
 using namespace detail;
 
 SCENARIO("IMSI range should default to Telefonica as Network Provider", "[cellular]") {
-    REQUIRE(_cellular_imsi_to_network_provider(nullptr) == CELLULAR_NETPROV_TELEFONICA);
-    REQUIRE(_cellular_imsi_to_network_provider("")   == CELLULAR_NETPROV_TELEFONICA);
-    REQUIRE(_cellular_imsi_to_network_provider("123456789012345") == CELLULAR_NETPROV_TELEFONICA);
-    REQUIRE(_cellular_imsi_to_network_provider("2040")  == CELLULAR_NETPROV_TELEFONICA);
-    REQUIRE(_cellular_imsi_to_network_provider("31041") == CELLULAR_NETPROV_TELEFONICA);
-    REQUIRE(_cellular_imsi_to_network_provider("2140")  == CELLULAR_NETPROV_TELEFONICA);
-    REQUIRE(_cellular_imsi_to_network_provider("0404")  == CELLULAR_NETPROV_TELEFONICA);
-    REQUIRE(_cellular_imsi_to_network_provider("10410") == CELLULAR_NETPROV_TELEFONICA);
-    REQUIRE(_cellular_imsi_to_network_provider("1407")  == CELLULAR_NETPROV_TELEFONICA);
+    REQUIRE(_cellular_sim_to_network_provider(nullptr, nullptr) == CELLULAR_NETPROV_TELEFONICA);
```

Comment:
Add unit tests for Gen 3 implementation? They can be instead added to our new CMake-based test suite. I can work with you next week on helping add those.

Reply:
thank u so much for helping me add the unit tests! @avtolstoy  :)